src/diffusers/models/transformers/auraflow_transformer_2d.py

```diff
@@ -253,6 +253,7 @@ class AuraFlowTransformer2DModel(ModelMixin, ConfigMixin):
         pos_embed_max_size (`int`, defaults to 4096): Maximum positions to embed from the image latents.
     """
 
+    _no_split_modules = ["AuraFlowJointTransformerBlock", "AuraFlowSingleTransformerBlock", "AuraFlowPatchEmbed"]
```

Comment:
I'm curious what's expected behavior of `_no_split_modules` here? 

I assume we would not split these specified layers across different devices, did I understand wrong? 

I think we are splitting the `AuraFlowSingleTransformerBlock` based on the device map  cc @SunMarc 

```
 hf_device_map:
{'register_tokens': 0, 'pos_embed': 0, 'context_embedder': 0, 'time_step_embed': 0, 'time_step_proj': 0, 
'joint_transformer_blocks': 0, 'single_transformer_blocks.0': 0, 'single_transformer_blocks.1': 0, 'single_transformer_blocks.2': 0, 'single_transformer_blocks.3': 0, 'single_transformer_blocks.4': 0, 
'single_transformer_blocks.5': 0, 'single_transformer_blocks.6': 0, 'single_transformer_blocks.7': 0, 
'single_transformer_blocks.8': 0, 'single_transformer_blocks.9': 0, 'single_transformer_blocks.10': 0, 
'single_transformer_blocks.11': 0, 'single_transformer_blocks.12': 'disk', 'single_transformer_blocks.13': 'disk', 'single_transformer_blocks.14': 'disk', 'single_transformer_blocks.15': 'disk', 
'single_transformer_blocks.16': 'disk', 'single_transformer_blocks.17': 'disk', 
'single_transformer_blocks.18': 'disk', 'single_transformer_blocks.19': 'disk', 
'single_transformer_blocks.20': 'disk', 'single_transformer_blocks.21': 'disk', 
'single_transformer_blocks.22': 'disk', 'single_transformer_blocks.23': 'disk', 
'single_transformer_blocks.24': 'disk', 'single_transformer_blocks.25': 'disk', 
'single_transformer_blocks.26': 'disk', 'single_transformer_blocks.27': 'disk', 
'single_transformer_blocks.28': 'disk', 'single_transformer_blocks.29': 'disk', 
'single_transformer_blocks.30': 'disk', 'single_transformer_blocks.31': 'disk',
 'norm_out': 'disk', 'proj_out': 'disk'}
```

Reply:
`_no_split_modules` are the modules that contain residual connections. As we can see that the blocks specified here -- all of them contain residual connections. Hence we cannot split. But I am happy to stand corrected @SunMarc.

Reply:
@yiyixuxu I think Marc agrees to what I said here. Could you take another look?

Reply:
sorry this PR somehow got missed - approved